test/jdk/java/net/spi/InetAddressResolverProvider/RuntimePermissionTest.java

```diff
@@ -85,11 +85,14 @@ public TestSecurityManager(boolean permitInetAddressResolver) {
         @Override
         public void checkPermission(Permission permission) {
             if (permission instanceof RuntimePermission) {
-                LOGGER.info("Checking RuntimePermission: " + permission);
                 if (RUNTIME_PERMISSION_NAME.equals(permission.getName()) && !permitInetAddressResolver) {
                     LOGGER.info("Denying '" + RUNTIME_PERMISSION_NAME + "' permission");
```

Comment:
Hello Aleksey, wouldn't this logger statement have the same issue as the one that was outside the if block? Should this logger message be removed too?

Reply:
Nope, because this one is gated by `RUNTIME_PERMISSION_NAME`, which is `inetAddressResolverProvider`. This does not form the cycle. It is the other path that gets us in trouble.

Reply:
> Nope, because this one is gated by RUNTIME_PERMISSION_NAME, which is inetAddressResolverProvider. This does not form the cycle.

I see what you mean and that makes sense. The change looks fine to me.